### PR TITLE
Improved Scribble Integration

### PR DESCRIPTION
**Pull request by dmuhs**
This PR will:
- refactor common scribble logic into an easy-to-use middleware class that can be used in solc- and truffle-related projects alike
- use the improved scribble JSON i/o logic
- add improved scribble tests by mocking out subprocess calls
- fix a bug where black was not correctly called locally due to a version upgrade of `tox`

### CODE REVIEW

File: mythx_cli/analyze/solidity.py [hunk 1]
```diff
@@ -191,48 +199,55 @@ def generate_payloads(
...
+        #
+        #     # don't delete temp file on close but manually unlink
+        #     # after payload has been generated
+        #     scribble_output_f = tempfile.NamedTemporaryFile(
+        #         mode="w+", delete=False, suffix=".sol"
+        #     )
+        #     scribble_stdout = process.stdout.decode()
+        #     scribble_output_f.write(scribble_stdout)
+        #     scribble_file = scribble_output_f.name
+        #     scribble_output_f.close()
```
Comment (joaosantos15) [inline mythx_cli/analyze/solidity.py:244]: Should this commented section be included?

File: mythx_cli/analyze/solidity.py [hunk 2]
```diff
@@ -254,19 +269,21 @@ def generate_payloads(
...
+        #         "scribble-" + str(self.target) if item == scribble_file else item
+        #         for item in payload["source_list"]
+        #     ]
+        #     payload["main_source"] = "scribble-" + str(self.target)
+        #
+        #     # delete scribble temp file
+        #     os.unlink(scribble_file)
+
+        # print(json.dumps(payload))
+        # sys.exit()
```
Comment (joaosantos15) [inline mythx_cli/analyze/solidity.py:286]: ⬆️ comments :D

**Review by joaosantos15 — APPROVED** [review]

File: mythx_cli/analyze/solidity.py (hunk 2 above)
Comment (dmuhs) [reply mythx_cli/analyze/solidity.py]: Fixed in 33bf358

File: mythx_cli/analyze/solidity.py (hunk 1 above)
Comment (dmuhs) [reply mythx_cli/analyze/solidity.py]: Fixed in 33bf358

**Pull request merged**